From 10ee300f246f835c53dddd75330caaa43d5484f4 Mon Sep 17 00:00:00 2001
From: Electronic Eel <electroniceel2@googlemail.com>
Date: Fri, 4 Jan 2019 19:13:39 +0100
Subject: [PATCH 1/3] Create initial vocore2-breakout config

Copy of the vocore2 without any changes except board name
---
 .../ramips/base-files/etc/board.d/01_leds     |  3 +-
 .../ramips/base-files/etc/board.d/02_network  |  1 +
 target/linux/ramips/base-files/etc/diag.sh    |  3 +-
 target/linux/ramips/base-files/lib/ramips.sh  |  3 +
 .../ramips/base-files/lib/upgrade/platform.sh |  1 +
 target/linux/ramips/dts/VOCORE2BREAKOUT.dts   | 55 +++++++++++++++++++
 target/linux/ramips/image/mt76x8.mk           |  9 +++
 7 files changed, 73 insertions(+), 2 deletions(-)
 create mode 100644 target/linux/ramips/dts/VOCORE2BREAKOUT.dts

diff --git a/target/linux/ramips/base-files/etc/board.d/01_leds b/target/linux/ramips/base-files/etc/board.d/01_leds
index 6057275978..5ae7eac8ce 100755
--- a/target/linux/ramips/base-files/etc/board.d/01_leds
+++ b/target/linux/ramips/base-files/etc/board.d/01_leds
@@ -446,7 +446,8 @@ vocore-16M)
 	ucidef_set_led_netdev "eth" "ETH" "vocore:orange:eth" "eth0"
 	set_wifi_led "vocore:green:status"
 	;;
-vocore2)
+vocore2|\
+vocore2breakout)
 	set_wifi_led "$boardname:fuchsia:status"
 	;;
 vocore2lite)
diff --git a/target/linux/ramips/base-files/etc/board.d/02_network b/target/linux/ramips/base-files/etc/board.d/02_network
index ebf40ad1fa..2c8bebaf3e 100755
--- a/target/linux/ramips/base-files/etc/board.d/02_network
+++ b/target/linux/ramips/base-files/etc/board.d/02_network
@@ -347,6 +347,7 @@ ramips_setup_interfaces()
 			"0:lan" "4:wan" "6@eth0"
 		;;
 	vocore2|\
+	vocore2breakout|\
 	vocore2lite)
 		ucidef_add_switch "switch0" \
 			"0:lan" "2:lan" "6t@eth0"
diff --git a/target/linux/ramips/base-files/etc/diag.sh b/target/linux/ramips/base-files/etc/diag.sh
index 2f51add331..caf9a92be7 100644
--- a/target/linux/ramips/base-files/etc/diag.sh
+++ b/target/linux/ramips/base-files/etc/diag.sh
@@ -279,7 +279,8 @@ get_status_led() {
 	vocore-16M)
 		status_led="vocore:green:status"
 		;;
-	vocore2)
+	vocore2|\
+	vocore2breakout)
 		status_led="$boardname:fuchsia:status"
 		;;
 	vocore2lite)
diff --git a/target/linux/ramips/base-files/lib/ramips.sh b/target/linux/ramips/base-files/lib/ramips.sh
index 5741cbd2ee..cc563907df 100755
--- a/target/linux/ramips/base-files/lib/ramips.sh
+++ b/target/linux/ramips/base-files/lib/ramips.sh
@@ -538,6 +538,9 @@ ramips_board_detect() {
 	*"VoCore2-Lite")
 		name="vocore2lite"
 		;;
+	*"VoCore2-Breakout")
+		name="vocore2breakout"
+		;;
 	*"VR500")
 		name="vr500"
 		;;
diff --git a/target/linux/ramips/base-files/lib/upgrade/platform.sh b/target/linux/ramips/base-files/lib/upgrade/platform.sh
index ffdc5e73e0..c133883ee2 100755
--- a/target/linux/ramips/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ramips/base-files/lib/upgrade/platform.sh
@@ -164,6 +164,7 @@ platform_check_image() {
 	vocore-16M|\
 	vocore2|\
 	vocore2lite|\
+	vocore2breakout|\
 	vr500|\
 	w150m|\
 	w2914nsv2|\
diff --git a/target/linux/ramips/dts/VOCORE2BREAKOUT.dts b/target/linux/ramips/dts/VOCORE2BREAKOUT.dts
new file mode 100644
index 0000000000..9d7c00e78a
--- /dev/null
+++ b/target/linux/ramips/dts/VOCORE2BREAKOUT.dts
@@ -0,0 +1,55 @@
+/dts-v1/;
+
+#include "VOCORE2.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "vocore,vocore2breakout", "vocore,vocore2", "mediatek,mt7628an-soc";
+	model = "VoCore2-Breakout";
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		status {
+			label = "vocore2breakout:fuchsia:status";
+			gpios = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+		m25p,chunked-io = <32>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x30000>;
+			read-only;
+		};
+
+		partition@30000 {
+			label = "u-boot-env";
+			reg = <0x30000 0x10000>;
+			read-only;
+		};
+
+		factory: partition@40000 {
+			label = "factory";
+			reg = <0x40000 0x10000>;
+			read-only;
+		};
+
+		partition@50000 {
+			label = "firmware";
+			reg = <0x50000 0xfb0000>;
+		};
+	};
+};
diff --git a/target/linux/ramips/image/mt76x8.mk b/target/linux/ramips/image/mt76x8.mk
index 21c5357a08..f12c48c651 100644
--- a/target/linux/ramips/image/mt76x8.mk
+++ b/target/linux/ramips/image/mt76x8.mk
@@ -241,6 +241,15 @@ define Device/vocore2
 endef
 TARGET_DEVICES += vocore2
 
+define Device/vocore2breakout
+  DTS := VOCORE2BREAKOUT
+  IMAGE_SIZE := $(ralink_default_fw_size_16M)
+  DEVICE_TITLE := VoCore VoCore2-Breakout
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-usb-ledtrig-usbport \
+    kmod-sdhci-mt7620
+endef
+TARGET_DEVICES += vocore2breakout
+
 define Device/vocore2lite
   DTS := VOCORE2LITE
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
-- 
2.17.2

